nengo_dl/simulator.py

```diff
@@ -1180,30 +1178,33 @@ def save_params(self, path, include_non_trainable=False):
         `.get_nengo_params`.
         """
 
-        vars = (
-            self.keras_model.weights
-            if include_non_trainable
-            else self.keras_model.trainable_weights
-        )
+        if include_non_trainable is not None:
+            warnings.warn(
+                "include_non_trainable is deprecated, use include_state instead",
+                DeprecationWarning,
+            )
+            include_state = include_non_trainable
+
+        vars = self.keras_model.weights
+        if include_state:
+            vars.extend(self.tensor_graph.saved_state.values())
```

Comment:
This will modify `self.keras_model.weights`, right? Is that a problem?

Reply:
Yeah good call probably safer to do this on a copy